python/tests/integration/planning/stepwise_planner/test_stepwise_planner.py

```diff
@@ -0,0 +1,180 @@
+from semantic_kernel.connectors.search_engine import BingConnector
+from semantic_kernel.core_skills.math_skill import MathSkill
+from semantic_kernel.core_skills.time_skill import TimeSkill
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.orchestration.sk_context import SKContext
+from semantic_kernel.planning import StepwisePlanner
+from semantic_kernel.planning.stepwise_planner.stepwise_planner_config import (
+    StepwisePlannerConfig,
+)
+from semantic_kernel.skill_definition import sk_function, sk_function_context_parameter
+
+
+class TempWebSearchEngineSkill:
+    """
+    TODO: replace this class with semantic_kernel.core_skills.web_search_engine_skill.WebSearchEngineSkill
+
+    SKFunction.describe() does not contains info for arguments.
+
+    so that `query: str` is not shown in the function description,
+    BUT this argument must be passed to planner to work appropriately.
+
+    This function temporarily add `query` as parameter by using @sk_function_context_parameter.
+    original file is here: semantic-kernel/python/semantic_kernel/core_skills/web_search_engine_skill.py
+    """
+
+    def __init__(self, connector) -> None:
+        self._connector = connector
+
+    @sk_function(
+        description="Performs a web search for a given query", name="searchAsync"
+    )
+    @sk_function_context_parameter(
+        name="query",
+        description="The search query",
+    )
+    async def search_async(self, query: str, context: SKContext) -> str:
+        query = query or context.variables.get("query")[1]
+        result = await self._connector.search_async(query, num_results=1, offset=0)
+        return str(result)
+
+
+@pytest.fixture(scope="session")
+def get_bing_config():
+    if "Python_Integration_Tests" in os.environ:
+        api_key = os.environ["Bing__ApiKey"]
+    else:
+        # Load credentials from .env file
+        api_key = sk.bing_search_settings_from_dot_env()
+
+    return api_key
+
+
+def initialize_kernel(get_aoai_config, use_embeddings=False, use_chat_model=False):
+    _, api_key, endpoint = get_aoai_config
+
+    kernel = Kernel()
+    if use_chat_model:
+        kernel.add_chat_service(
+            "chat_completion",
+            sk_oai.AzureChatCompletion("gpt-35-turbo", endpoint, api_key),
+        )
+    else:
+        kernel.add_text_completion_service(
+            "text_completion",
+            sk_oai.AzureChatCompletion("gpt-35-turbo", endpoint, api_key),
+        )
+
+    if use_embeddings:
+        kernel.add_text_embedding_generation_service(
+            "text_embedding",
+            sk_oai.AzureTextEmbedding("text-embedding-ada-002", endpoint, api_key),
+        )
+    return kernel
+
+
+@pytest.mark.parametrize(
+    "use_chat_model, prompt, expected_function, expected_skill",
+    [
+        (
+            False,
+            "Who is the current president of the United States? What is his current age divided by 2",
```

Comment:
I believe it makes more sense to run an integration test against a fact that is permanent - this test will have to be changed at some point otherwise. I also find that this test fails for me regularly on my machine.

Reply:
Maybe "How long was Henry VIII king of England? How many times was he married divided by 2?"

Reply:
Update: the above prompt works well with the tests for me

Reply:
@awharrison-28  I think your case is much better ! I'll update it :)

Reply:
Doing similar in #2550 2550